services/sleeper/tests/integration/test_docker_container.py

```diff
@@ -0,0 +1,205 @@
+                "bind": container_variables["{}_FOLDER".format(str(folder).upper())]
+            }
+            for folder in _FOLDER_NAMES
+        }
+        container = docker_client.containers.run(
+            docker_image_key,
+            "run",
+            detach=True,
+            remove=False,
+            volumes=volumes,
+            environment=container_variables,
+        )
+        response = container.wait()
+        if response["StatusCode"] > 0:
+            logs = container.logs(timestamps=True)
+            pytest.fail(
+                "The container stopped with exit code {}\n\n\ncommand:\n {}, \n\n\nlog:\n{}".format(
+                    response["StatusCode"],
+                    "run",
+                    pformat(
+                        (container.logs(timestamps=True).decode("UTF-8")).split("\n"),
+                        width=200,
+                    ),
+                )
+            )
+        else:
+            yield container
+    except docker.errors.ContainerError as exc:
+        # the container did not run correctly
+        pytest.fail(
+            "The container stopped with exit code {}\n\n\ncommand:\n {}, \n\n\nlog:\n{}".format(
+                exc.exit_status,
+                exc.command,
+                pformat(
+                    (container.logs(timestamps=True).decode("UTF-8")).split("\n"),
+                    width=200,
+                ),
+            )
+        )
+    finally:
+        # cleanup
+        container.remove()
+
+
+def _convert_to_simcore_labels(image_labels: Dict) -> Dict:
+    io_simcore_labels = {}
+    for key, value in image_labels.items():
+        if str(key).startswith("io.simcore."):
+            simcore_label = json.loads(value)
+            simcore_keys = list(simcore_label.keys())
+            assert len(simcore_keys) == 1
+            simcore_key = simcore_keys[0]
+            simcore_value = simcore_label[simcore_key]
+            io_simcore_labels[simcore_key] = simcore_value
+    assert len(io_simcore_labels) > 0
+    return io_simcore_labels
+
+
+def test_run_container(
+    validation_folders: Dict,
+    host_folders: Dict,
+    docker_container: docker.models.containers.Container,
+):
+    for folder in _FOLDER_NAMES:
+        # test if the files that should be there are actually there and correct
+        list_of_files = [
+            x.name
+            for x in validation_folders[folder].iterdir()
+            if not ".gitkeep" in x.name
+        ]
+        for file_name in list_of_files:
+            assert Path(
+                host_folders[folder] / file_name
+            ).exists(), "missing files in {}".format(host_folders[folder])
+        match, mismatch, errors = filecmp.cmpfiles(
+            host_folders[folder],
+            validation_folders[folder],
+            list_of_files,
+            shallow=False,
+        )
+        # assert not mismatch, "wrong/incorrect files in {}".format(host_folders[folder])
```

Comment:
@KZzizzle that is why the test does not fail even if the contents are different. But I agree this should be made better. I'll add an issue in the cookiecutter https://github.com/ITISFoundation/cookiecutter-osparc-service/issues/27

Reply:
Do I need change something here?

Reply:
no that is the idea
basically it is up to the service provider to also provide tests for testing the contents